### ISSUE TITLE
strange pixel issues on raspberry pi4

### ISSUE THREAD

**Issue by shanbady** [Windblade-GR01/Ikemen-GO#303]
I finally got this work on a raspberry pi - however one thing I noticed is that certain things have very strange color and pixel issues. Certain characters have funky colors and certain interface elements appear oddly pixelated and/or off color. This is most immediately noticeable on the splash screen. 

![sample](https://user-images.githubusercontent.com/196425/125552819-456c9d75-d484-4867-97e5-937faf247284.png)

is this due to the some default os color pallete on the raspberry pi (it's a raspberry pi4 running the ubuntu desktop) or is this an issue with Ikemen on linux?

I'm just familiarizing myself with the source code to track down what I can play with to fix it myself but I'm a bit lost.

**Comment by shanbady** [comment]
I wonder if this is somehow related to https://github.com/Windblade-GR01/Ikemen-GO/issues/288

**Comment by SuperFromND** [comment]
I don't believe this is related to that issue, as that's just an issue with one specific state controller.  This seems like a more fundamental issue regarding how the Pi's GLES implementation handles whatever code IKEMEN uses to handle .sff palettes.  I sadly don't know the specifics beyond that, as I'm not familiar with IKEMEN's source or the nature of GLES.

**Comment by joamjoamjoam** [comment]
Fixed in commit 86e59c1a7cb619239e67caf3950765fde7b618c6:

The texture mapping in the fragment shaders was off slightly due to different floating point precision of the rPi Broadcom CPU

@Windblade-GR01 if your satisfied with that commit feel free to close this.
